### ISSUE TITLE
Failed to build and run the tests

### ISSUE THREAD

**Issue by nene** [benjie/prettier-plugin-pg#25]
I wanted to try out this plugin, but failed to build it locally on MacOS Monterey.

First off I was missing Python. Then it probably wasn't happy with the version of Python:

```
gyp ERR! configure error 
gyp ERR! stack Error: Command failed: /usr/bin/python3 -c import sys; print "%s.%s.%s" % sys.version_info[:3];
gyp ERR! stack   File "<string>", line 1
gyp ERR! stack     import sys; print "%s.%s.%s" % sys.version_info[:3];
gyp ERR! stack                       ^
gyp ERR! stack SyntaxError: invalid syntax
```

Looks like the problem stems from `pg-query-native-latest` which uses `node-gyp` 3.8.0.

The [docs for node-gyp][gyp] say that there are problems with running it on MacOS Catalina or later and the advised solution is to upgrade to node-gyp >= 7.

I tried swapping `pg-query-native-latest 1.1.1` with `pg-query-native 1.3.1` which uses node-gyp 8.2 (also renaming `typings/pg-query-native-latest.d.ts` to `typings/pg-query-native.d.ts` and changing references to `pg-query-native-latest` elsewhere). This did fix the installation. I was also able to compile the code with tsc. But the tests failed to run, with error messages like:

```
 FAIL   test  tests/BEGIN/jsfmt.spec.js
  ● Test suite failed to run

    Node doesn't have location:

    {
      TransactionStmt: {
        kind: 0,
        options: [
          {
            DefElem: {
              defname: 'transaction_deferrable',
              arg: { A_Const: [Object] },
              defaction: 0
            }
          }
        ]
      }
    }

    Parent node:

    {
      Document: {
        statements: [
          { TransactionStmt: { kind: 0, options: [Array] } },
          { TransactionStmt: { kind: 0, options: [Array] } }
        ],
        doc_location: 0,
        doc_len: 50
      },
      comments: [ { LineComment: true, value: '-- Hello!', start: 0, end: 9 } ],
      start: 0,
      end: 50
    }
```

[gyp]: https://github.com/nodejs/node-gyp/blob/HEAD/macOS_Catalina.md

**Comment by nene** [comment]
I also tried installing the 0.0.0 version of the library through NPM. This failed with the same error.

**Comment by nene** [comment]
My interest in this plugin stems from the fact that I'm actually writing an [SQL-formatting Prettier plugin][cst-plugin] by myself. So I was mainly looking for some inspiration, mostly just wanting to see what kind of formatting this plugin produces.

From the code in `print.ts` I see e.g. some formatting of SELECT statement, but when I look at the tests, the formatting seems to be very rudimentary. More likely though I'm just witnessing lack of tests.

I read from another issue in this repo that you've been having trouble finding a good SQL parser to use. That's also been the problem I've been facing, and which drove me to eventually write an [SQL parser][cst-parser] on my own. My parser outputs a CST (rather than AST), which so far has proven itself as a good choice for implementing formatting logic on top of.

Unfortunately my parser doesn't yet support PostgreSQL. While I do have a plan of implementing it, that will take some time... Postgres is kind of big :P

[cst-plugin]: https://github.com/nene/prettier-plugin-sql-cst
[cst-parser]: https://github.com/nene/sql-parser-cst

**Comment by benjie** [comment]
> the formatting seems to be very rudimentary. More likely though I'm just witnessing lack of tests.

Nope; it's extremely rudimentary - in fact it doesn't work for many statements at all. We very quickly hit a wall with the Postgres parser - insufficient location data and the stripping of comments meant that we couldn't re-attach comments in the right places. Other parsers have other issues, and I've yet to find the one that'll solve them all. I guess from your comments you've already read https://github.com/benjie/prettier-plugin-pg/issues/10#issuecomment-701675765

I actually started writing a parser for Postgres too, but honestly even when I attempt to automate chunks of it it's still a huge undertaking. So this project remains on my backburner.

Good luck!

**Comment by nene** [comment]
Thanks for the info.
